src/main/java/uk/gov/hmcts/reform/refunds/services/RefundsServiceImpl.java

```diff
@@ -100,84 +102,69 @@ public RefundResponse initiateRefund(RefundRequest refundRequest, MultiValueMap<
-        Set<String> distintUserIDSet = idamService.getUserIdSetForRoles(headers, roles);
-
-        if (optionalRefundList.isPresent()) {
-            List<Refund> refundList = optionalRefundList.get();
-            refundList = refundList.stream()
-                    .filter(refunds -> distintUserIDSet.contains(refunds.getCreatedBy()))
-                    .collect(Collectors.toList());
-            return Optional.of(refundList);
-        }
-        throw new RefundListEmptyException("Refund list is empty for given criteria");
-    }
-
-    public RefundListDtoResponse getRefundListDto(MultiValueMap<String, String> headers, Optional<List<Refund>> refundList) {
+    public RefundListDtoResponse getRefundListDto(MultiValueMap<String, String> headers, Optional<List<Refund>> refundList, List<String> roles) {
 
         if (refundList.isPresent() && !refundList.get().isEmpty()) {
             return RefundListDtoResponse
                 .buildRefundListWith()
-                .refundList(getRefundResponseDtoList(headers, refundList.get()))
+                .refundList(getRefundResponseDtoList(headers, refundList.get(), roles))
                 .build();
         } else {
             throw new RefundListEmptyException("Refund list is empty for given criteria");
         }
     }
 
-    public List<RefundDto> getRefundResponseDtoList(MultiValueMap<String, String> headers, List<Refund> refundList) {
-        //Distinct createdBy UID
-        Set<String> distintUIDSet = refundList
-            .stream().map(Refund::getCreatedBy)
-            .collect(Collectors.toSet());
-
-        //Map UID -> User full name
-        Map<String, UserIdentityDataDto> userFullNameMap = getIdamUserDetails(headers, distintUIDSet);
+    public List<RefundDto> getRefundResponseDtoList(MultiValueMap<String, String> headers, List<Refund> refundList, List<String> roles) {
 
         //Create Refund response List
         List<RefundDto> refundListDto = new ArrayList<>();
 
-        //Update the user full name for created by
-        refundList
-            .forEach(refund -> {
-                String reason  = getRefundReason(refund.getReason());
-                refundListDto.add(refundResponseMapper.getRefundListDto(
-                    refund,
-                    userFullNameMap.get(refund.getCreatedBy()),reason
-                ));
-                     }
-
-
-            );
+        if (!roles.isEmpty()) {
+            List<UserIdentityDataDto> userIdentityDataDtoList = idamService.getUsersForRoles(headers, roles);
+
+            // Filter Refunds List based on Refunds Roles and Update the user full name for created by
+            refundList.stream()
+                    .filter(e -> userIdentityDataDtoList.stream().map(UserIdentityDataDto::getId)
+                            .anyMatch(id -> id.equals(e.getCreatedBy())))
+                    .collect(Collectors.toList())
+                    .forEach(refund -> {
```

Comment:
can this logic be handled in the above stream?

Reply:
No, as it returns dto list